capstone/src/main/java/com/google/sps/Objects/Post.java

```diff
@@ -46,15 +49,19 @@ public String getImg() {
     return img;
   }
 
-  public ArrayList<User> getLikes() {
+  public HashSet<String> getLikes() {
     return likes;
   }
 
   public void addComment(Comment newComment) {
     this.comments.add(newComment);
   }
 
-  public void addNewLike(User newLike) {
-    this.likes.add(newLike);
+  public void addLike(String userId) {
+    if (!likes.contains(userId)) this.likes.add(userId);
+  }
+
+  public void removeLike(String userId) {
```

Comment:
I think if you remove something that doesn't exist in the set it'll raise an error so you might want to write in a check

Reply:
for HashSets I think it returns false if the item doesn't exist